src/main/java/org/truffleruby/language/dispatch/CachedDispatchNode.java

```diff
@@ -139,14 +141,11 @@ protected final boolean guardName(Object methodName) {
 
         if (cachedName instanceof String) {
             return cachedName.equals(methodName);
-        } else if (cachedName == cachedNameAsSymbol) { // If cachedName is a Symbol
-            // The above guard already proved they are different
-            return false;
         } else if (cachedNameIsRubyString) {
             return RubyGuards.isRubyString(methodName) && StringOperations.rope((DynamicObject) cachedName).equals(StringOperations.rope((DynamicObject) methodName));
-        } else {
-            CompilerDirectives.transferToInterpreterAndInvalidate();
-            throw new UnsupportedOperationException();
+        } else { // cachedName is a Symbol
+            // The above guard already proved they are different
```

Comment:
Might be worth clarifying this comment here. I think the logic is that if cachedName is a Symbol that matched then `cachedName == methodName` would have been true, and since it wasn't the result must be false?